### PR TITLE
Docs/Tracing: Add sections about schema of the trace data

### PR DESCRIPTION
**Pull request by aocenas**
Adds a section explaining how the tracing data should be send from a data source so that it can be displayed in the trace view.

### CODE REVIEW

File: docs/sources/explore/trace-integration.md [hunk 1]
```diff
@@ -66,3 +66,36 @@ You can navigate from a span in a trace view directly to logs relevant for that
...
+
+| Field name | Type    | Description | 
+|------------|---------|-------------|
+| logs       | TraceLog[] | List of logs associated with the current span. |
+| tags       | TraceKeyValuePair[]  | List of tags associated with the current span. |
+| warnings   | string[]  | List of warnings associated with the current span. |
+| stackTraces | string[] | List of stack traces associated with the current span. |
+| errorIconColor | string | Color of the error icon in case span is tagged with `error: true`. |
+
+For details about the types see [TraceSpanRow](https://grafana.com/docs/grafana/latest/packages_api/data/tracespanrow/), [TraceKeyValuePair](https://grafana.com/docs/grafana/latest/packages_api/data/tracekeyvaluepair/) and [TraceLog](https://grafana.com/docs/grafana/latest/packages_api/data/tracelog/)
```
Comment (aocenas) [inline docs/sources/explore/trace-integration.md:101]: @dprokop is there a way to link this relatively? Also not sure how to build those docs locally so did not check it.

Comment (oddlittlebird) [reply docs/sources/explore/trace-integration.md:101]: @mckn, can you help?

**Comment by oddlittlebird** [comment]
@achatterjee-grafana, the docs are still Grafana 7.4 at this exact moment. If it's merged today, we might want to add a version note.

File: docs/sources/explore/trace-integration.md [hunk 2]
```diff
@@ -66,3 +66,36 @@ You can navigate from a span in a trace view directly to logs relevant for that
...
+
+Required fields:
+
+| Field name | Type    | Description | 
+|------------|---------|-------------|
+| traceID    | string  | Identifier for the entire trace. There should be only one trace in the data frame. |
+| spanID     | string  | Identifier for the current span. SpanIDs should be unique per trace. |
+| parentSpanID | string  | SpanID of the parent span to create child parent relationship in the trace view. Can be `undefined` for root span without parent. |
+| serviceName | string  | Name of the service this span is part of. |
+| serviceTags | TraceKeyValuePair[] | List of tags relevant for the service. |
```
Comment (zoltanbedi) [inline docs/sources/explore/trace-integration.md:86]: serviceTags is in the table 2 times

**Review by zoltanbedi — APPROVED** [review]
other than the serviceTags, this looks good to me.

**Comment by achatterjee-grafana** [comment]
Let us hold off from a merge. @oddlittlebird  is working with Will to update the doc version for the website. Tomorrow is fine, and thanks for your patience and understanding!

**Review by achatterjee-grafana — APPROVED** [review]

File: docs/sources/explore/trace-integration.md (hunk 1 above)
Comment (achatterjee-grafana) [reply docs/sources/explore/trace-integration.md:101]: The packages_api docs are not in the grafana docs repo. Where do they stay? Thanks for any clarification.

Comment (oddlittlebird) [reply docs/sources/explore/trace-integration.md:101]: The docs do not exist until build time. Only the source documentation exists. That is about all I know.

Comment (mckn) [reply docs/sources/explore/trace-integration.md:100]: Yes, you can link to it by using the following syntax `@link`. 

Here you have an example: https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/datetime/common.ts#L36

And here you have some more docs: https://api-extractor.com/pages/tsdoc/tag_link/

**Comment by oddlittlebird** [comment]
@aocenas, please add a link to this updated documentation to the What's new in Grafana 7.5 page. Thank you!

File: docs/sources/explore/trace-integration.md (hunk 1 above)
Comment (mckn) [reply docs/sources/explore/trace-integration.md:100]: Ah, mybad - I misunderstood the context. So making it the way you have done it is the only way I can think of. The downside is that is will break if we change anything in the code 🙈

**Pull request merged**